manta-crypto/src/eclair/cmp.rs

```diff
@@ -56,30 +60,155 @@ where
     }
 }
 
-/* TODO:
-impl<T, Rhs> PartialEq<Rhs> for T
+/// Implements [`PartialEq`] for the given `$type`.
+macro_rules! impl_partial_eq {
+    ($($type:tt),* $(,)?) => {
+        $(
+            impl<Rhs> PartialEq<Rhs> for $type
+            where
+                $type: cmp::PartialEq<Rhs>,
```

Comment:
Are you implementing PartialEq for type, where type: PartialEq? Isn't this doing nothing?

Reply:
Notice the types here. `PartialEq` is the trait from ECLAIR and `cmp::PartialEq` is the trait from the Rust standard library. This macro lifts an implementation from Rust standard library to ECLAIR. If this were actually the same trait it would not compile since it would just be an infinite regress.